pymc3/model.py

```diff
@@ -1563,132 +1503,6 @@ def __call__(self, *args, **kwargs):
-            end = []
-        if (len(begin) + len(end)) > ndim:
-            raise ValueError(
-                "Length of `total_size` is too big, "
-                "number of scalings is bigger that ndim, got %r" % total_size
-            )
-        elif (len(begin) + len(end)) == 0:
-            return floatX(1)
-        if len(end) > 0:
-            shp_end = shape[-len(end) :]
-        else:
-            shp_end = np.asarray([])
-        shp_begin = shape[: len(begin)]
-        begin_coef = [floatX(t) / shp_begin[i] for i, t in enumerate(begin) if t is not None]
-        end_coef = [floatX(t) / shp_end[i] for i, t in enumerate(end) if t is not None]
-        coefs = begin_coef + end_coef
-        coef = tt.prod(coefs)
-    else:
-        raise TypeError(
-            "Unrecognized `total_size` type, expected int or list of ints, got %r" % total_size
-        )
-    return tt.as_tensor(floatX(coef))
-
-
-class FreeRV(Factor, PyMC3Variable):
-    """Unobserved random variable that a model is specified in terms of."""
-
-    dshape = None  # type: Tuple[int, ...]
-    size = None  # type: int
-    distribution = None  # type: Optional[Distribution]
-    model = None  # type: Optional[Model]
-
-    def __init__(
-        self,
-        type=None,
-        owner=None,
-        index=None,
-        name=None,
-        distribution=None,
-        total_size=None,
-        model=None,
-    ):
-        """
-        Parameters
-        ----------
-        type: theano type (optional)
-        owner: theano owner (optional)
-        name: str
-        distribution: Distribution
-        model: Model
-        total_size: scalar Tensor (optional)
-            needed for upscaling logp
-        """
-        if type is None:
-            type = distribution.type
-        super().__init__(type, owner, index, name)
-
-        if distribution is not None:
-            self.dshape = tuple(distribution.shape)
-            self.dsize = int(np.prod(distribution.shape))
-            self.distribution = distribution
-            self.tag.test_value = (
-                np.ones(distribution.shape, distribution.dtype) * distribution.default()
-            )
-            self.logp_elemwiset = distribution.logp(self)
-            # The logp might need scaling in minibatches.
-            # This is done in `Factor`.
-            self.logp_sum_unscaledt = distribution.logp_sum(self)
-            self.logp_nojac_unscaledt = distribution.logp_nojac(self)
-            self.total_size = total_size
-            self.model = model
-            self.scaling = _get_scaling(total_size, self.shape, self.ndim)
-
-            incorporate_methods(
-                source=distribution,
-                destination=self,
-                methods=["random"],
-                wrapper=InstanceMethod,
-            )
-
-    @property
-    def init_value(self):
-        """Convenience attribute to return tag.test_value"""
-        return self.tag.test_value
-
-
 def pandas_to_array(data):
```

Comment:
Not related to this PR, but this function could probably benefit from a new name